### ISSUE TITLE
Setting empty string for alias parameter generates an error message

### ISSUE THREAD

**Issue by mpakoupete** [elastic/terraform-provider-ec#802]
## Readiness Checklist

- [ ] I am running the latest version
- [ ] I checked the documentation and found no answer
- [ ] I checked to make sure that this issue has not already been filed
- [ ] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior

Setting `alias=""` blows up the provider. The provider should likely modify that to `Unknown` during planning.

## Current Behavior

Setting `alias=""` or `alias= null` generates an error message like bellow

```error
(...)
c_deployment.example_minimal: Still creating... [2m40s elapsed]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to ec_deployment.example_minimal, provider "provider[\"registry.terraform.io/elastic/ec\"]" produced an unexpected new value: .alias:
│ was cty.StringVal(""), but now cty.StringVal("my-example-deployment").
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

## Steps to Reproduce

With any [sample](https://registry.terraform.io/providers/elastic/ec/latest/docs/resources/deployment#basic) terraform template by  adding an empty alias `alias=""` we get the error above.

**Comment by gigerdo** [comment]
There is also another problem that even if `alias=""` can be set, the setting is ignored. The `cloud-sdk-go` currently considers `alias=undefined` and `alias=""` to be the same, while the API actually has a different understanding:
- `alias=undefined` -> use default
- `alias=""` -> disable alias

**Comment by gigerdo** [comment]
Will be fixed with release 0.11.0
